packages/editor/src/components/media-categories/index.js

```diff
@@ -0,0 +1,182 @@
+	_title = getExternalLink( foreignLandingUrl, _title );
+	const fullLicense = getOpenverseLicense(
+		license,
+		licenseVersion,
+		licenseUrl
+	);
+	let _creator = decodeEntities( creator );
+	if ( creatorUrl ) {
+		_creator = getExternalLink( creatorUrl, creator );
+	}
+	_creator = sprintf(
+		/* translators: %s: Name of the media's creator. */
+		__( ' by %s' ),
+		_creator
+	);
+	const caption = `"${ _title }" ${ _creator }/ ${ fullLicense }.`;
+	return caption.replace( /\s{2}/g, ' ' );
+};
+
+const coreMediaFetch = async ( query = {} ) => {
+	const mediaItems = await resolveSelect( coreStore ).getMediaItems( {
+		...query,
+		orderBy: !! query?.search ? 'relevance' : 'date',
+	} );
+	return mediaItems.map( ( mediaItem ) => ( {
+		...mediaItem,
+		alt: mediaItem.alt_text,
+		url: mediaItem.source_url,
+		previewUrl: mediaItem.media_details?.sizes?.medium?.source_url,
+		caption: mediaItem.caption?.raw,
+	} ) );
+};
+
+const inserterMediaCategories = [
+	{
+		label: __( 'Images' ),
+		name: 'images',
+		mediaType: 'image',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'image' } );
+		},
+	},
+	{
+		label: __( 'Videos' ),
+		name: 'videos',
+		mediaType: 'video',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'video' } );
+		},
+	},
+	{
+		label: __( 'Audio' ),
+		name: 'audio',
+		mediaType: 'audio',
+		async fetch( query = {} ) {
+			return coreMediaFetch( { ...query, media_type: 'audio' } );
+		},
+	},
+	{
+		label: __( 'Openverse' ),
+		searchLabel: __( 'Openverse' ),
+		name: 'openverse',
+		mediaType: 'image',
+		async fetch( query = {} ) {
+			const defaultArgs = {
+				mature: false,
+				excluded_source: 'flickr,inaturalist,wikimedia',
+				license: 'pdm,cc0',
+			};
+			const finalQuery = { ...query, ...defaultArgs };
+			const mapFromInserterMediaRequest = {
+				per_page: 'page_size',
+				search: 'q',
+			};
+			const url = new URL(
+				'https://api.openverse.engineering/v1/images/'
+			);
+			Object.entries( finalQuery ).forEach( ( [ key, value ] ) => {
+				const queryKey = mapFromInserterMediaRequest[ key ] || key;
+				url.searchParams.set( queryKey, value );
+			} );
+			const response = await window.fetch( url, {
+				headers: {
+					'User-Agent': 'WordPress/inserter-media-fetch',
+				},
+			} );
+			const jsonResponse = await response.json();
+			const results = jsonResponse.results;
+			const withAdjustments = results.map( ( result ) => ( {
+				...result,
+				// This is a temp solution for better titles, until Openverse API
+				// completes the cleaning up of some titles of their upstream data.
+				title: result.title?.toLowerCase().startsWith( 'file:' )
+					? result.title.slice( 5 )
+					: result.title,
+			} ) );
+			return withAdjustments.map( ( result ) => ( {
+				...result,
+				sourceId: result.id,
+				id: undefined,
+				caption: getOpenverseCaption( result ),
+				previewUrl: result.thumbnail,
+			} ) );
```

Comment:
Minor point, but I find this sequence of anonymous maps a little awkward. I'd choose one of the following:

* merge the mappers into one:

```js
return jsonResponse.results.map( ( result ) => ( {
  ...result,
  title: …,
  sourceId: …,
  id: …,
  caption: …,
  previewUrl: …,
} )
```

* name the mappers properly to make the meaning and separation clear:

```js
return jsonResponse.results
  .map( removeFileSchemaPrefix )
  .map( prepareInserterMediaItem )
```

Given how little code there is, my personal preference goes to option 1.